packages/core/hooks/hook_common.py

```
@@ -296,6 +318,67 @@ def resolve_path_within(project_dir: str, relative: str, filename: str) -> str |
+    その親を root として採用する。``--separate-git-dir`` 等で common dir が
+    ``.git`` 以外の名前になる構成は、安全のため解決対象外として None を返す。
+    git が使えない・結果が不正な場合も None を返す
+    (呼び出し側は project_dir へのフォールバックに使うこと)。
+
+    Args:
+        project_dir: git を実行する作業ディレクトリ。省略時は CWD。
+        timeout: git コマンドのタイムアウト秒数。
+    """
+    try:
+        result = subprocess.run(
+            ["git", "rev-parse", "--path-format=absolute", "--git-common-dir"],
```

Ruff (0.15.21)

[error] Starting a process with a partial executable path

(S607)